### PR TITLE
Remove unused (?) HealthCheck from KubeletClient

### PR DESCRIPTION
**Pull request by gmarek**
Needed for #12912 

cc @dchen1107 @yujuhong @wojtek-t

### CODE REVIEW

File: test/e2e/proxy.go [hunk 1]
```diff
@@ -52,6 +52,7 @@ func proxyContext(version string) {
 	f := NewFramework("proxy")
 	prefix := "/api/" + version
 
+	// Port here has to be kept in sync with default kubelet port.
```
Comment (mikedanese) [inline test/e2e/proxy.go]: there is a const for this:

https://github.com/kubernetes/kubernetes/blob/master/pkg/master/ports/ports.go#L28

**Comment by yujuhong** [comment]
LGTM. Thanks for the cleanup!

File: test/e2e/proxy.go (hunk 1 above)
Comment (gmarek) [inline test/e2e/proxy.go]: I don't understand what you mean.

Comment (mikedanese) [inline test/e2e/proxy.go]: Kind of a nit/suggestion. What I mean is the next line could be:
```go
It("should proxy logs on node with explicit kubelet port", func() {
     nodeProxyTest(f, version, fmt.Sprintf(":%v/logs/", ports.KubeletPort))
})
```
Then no need for the comment and we only have one thing to keep in sync which is the pkg/master/ports constants.

Comment (gmarek) [inline test/e2e/proxy.go]: This will be no longer true and each Kubelet will be able to have different port.

**Pull request merged**